src/sagas/flavor.js

```diff
@@ -1,27 +1,25 @@
-import { all, call, put, takeLatest, select, take } from 'redux-saga/effects';
-
+import { all, call, put, takeLatest, select } from 'redux-saga/effects';
 import request from 'utils/request';
-import { getUser } from 'selectors/application';
+import helper from 'utils/saga';
+import { getCurrentUser } from 'sagas/profile';
+import { isLoaded } from 'selectors/flavor';
 import { actions, types } from 'reducers/flavor';
-import { actions as appActions, types as appTypes } from 'reducers/application';
 
 function* requestStashWorker() {
   try {
-    let user = yield select(getUser);
-
-    if (user === null) {
-      yield put(appActions.requestCurrentUser());
-      yield take([
-        appTypes.REQUEST_CURRENT_USER_SUCCESS,
-        appTypes.REQUEST_CURRENT_USER_FAILURE
-      ]);
-      user = yield select(getUser);
+    const loaded = yield select(isLoaded);
+
+    if (loaded) {
+      return true;
```

Comment:
I think this can just `return;`

Reply:
I had it return true, because I didn't like the test expecting `undefined`.  lol Removed it